### DISCUSSION TITLE
The protoc-gen-connect-web plugin is deprecated

### DISCUSSION THREAD

**Discussion by smaye81** [connectrpc/connect-es#647] [Announcements]
Hey everyone. Just wanted to drop a quick a FYI that we have deprecated the old `@bufbuild/protoc-gen-connect-web` plugin on NPM. It has been nominally deprecated for a while now via a notice on the README. However, it is now officially deprecated. Any further usage will show a warning upon install.

If you are still depending on this package, please switch to use [@bufbuild/protoc-gen-connect-es](https://www.npmjs.com/package/@bufbuild/protoc-gen-connect-es), which is the official Connect-ES plugin. The generated code is actually exactly the same, so you should not experience any issues updating

To make the switch:

```bash
npm remove @bufbuild/protoc-gen-connect-web
npm install @bufbuild/protoc-gen-connect-es
```

Update your `buf.gen.yaml`:

```diff
version: v1
plugins:
  - plugin: es
    out: src/gen
-  - plugin: connect-web
+  - plugin: connect-es
    out: src/gen
```

And your import paths:

```diff
- import { ElizaService } from "gen/eliza_connectweb";
+ import { ElizaService } from "gen/eliza_connect";
```

Any issues or questions, please let us know in this discussion. Thanks!

**Comment by timostamm** [comment]
Connect RPC has [joined the CNCF](https://buf.build/blog/connect-rpc-joins-cncf).

Starting with [v0.13.1](https://github.com/connectrpc/connect-es/releases/tag/v0.13.1), the scope for Connect packages changed from `@bufbuild` to `@connectrpc` - the more current npm commands to run are:

```
npm remove @bufbuild/protoc-gen-connect-web
npm install @connectrpc/protoc-gen-connect-es
```

Note that we've also released a migration tool to make this change and update imports for you: `npx @connectrpc/connect-migrate@latest` will upgrade to the latest release.
